### PR TITLE
[Offline] Offline mode. Now with Video!

### PR DESCRIPTION
**Pull request by Donhusum**
## Description

This PR adds all the offline abilities to videos, so that the downloaded lectures now have and can display videos when the user is offline. 

## Changes

- Added functionality to storeCourseLocally and deleteLocallyStoredCourse so they both include videos. 
- Added the function getVideoURL in StorageService, to determine where to get videos from (online or offline). 
- Replaced an asyncStorage function with one from StorageService 

## Related Issues

It uses features from PR #214 so some of the code changes might be identical to the changes in that PR 

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.

## Screenshots (if applicable)

Any screenshot would look like what you have already seen. The difference is, that now you can see it all while offline (if you have downloaded the course) 

## If mobile/frontend pull request, what version of the backend is it stable, and running on? 

Branch: dev

Commit id: 8cb56a _5f5ff46ea18ad34f6dfc54de5485c0f8c8_

## Notes for Reviewers

It uses features from PR #214 so some of the code changes might be identical to the changes in that PR

### CODE REVIEW

**Comment by Jatewo** [comment]
Are these not the same changes as in PR #214? Let's not make multiple PRs with the same changes. If you want to add changes to a PR, just push them to the branch. Which one is the real PR?

**Comment by Donhusum** [comment]
> Are these not the same changes as in PR #214? Let's not make multiple PRs with the same changes. If you want to add changes to a PR, just push them to the branch. Which one is the real PR?

They are not, but the changes in this PR are depended on the once in the other PR, and I choose not to wait for it to be in dev.

**Review by PrometheusBork — APPROVED** [review]
Tested locally on my device and it works

File: eml/screens/lectures/VideoLectureScreen.jsx [hunk 1]
```diff
@@ -116,9 +117,9 @@ export default function VideoLectureScreen({ lectureObject, courseObject, isLast
   };
 
   useEffect(() => {
-    const _videoUrl = getVideoStreamUrl(lectureObject.video, currentResolution);
-
-    setVideoUrl(_videoUrl);
+    getVideoURL(lectureObject.video, currentResolution).then((result)=>{
+      setVideoUrl(result);
+    }); //getVideoStreamUrl(lectureObject.video, currentResolution); //HERE
```
Comment (ITPini) [inline eml/screens/lectures/VideoLectureScreen.jsx:122]:
```suggestion
    });
```

File: eml/services/StorageService.js [hunk 2]
```diff
@@ -325,65 +333,36 @@ export const fetchLectureImage = async (imageID, lectureID) => {
...
     if (isOnline) {
-      exerciseList = await api.getExercisesInSection(sectionID);
+      videoUrl = api.getVideoStreamUrl(videoName, resolution);
     } else {
-      throw new Error('No internet connection in getExerciseList');
+      throw new Error('No internet connection.');
     }
-  } catch (error) {
-    // Use locally stored exercises if they exist and the DB cannot be reached
+  } catch (unusedErrorMessage) {
```
Comment (ITPini) [inline eml/services/StorageService.js:353]:
```suggestion
  } catch (error) {
```

File: eml/services/StorageService.js [hunk 3]
```diff
@@ -325,65 +333,36 @@ export const fetchLectureImage = async (imageID, lectureID) => {
...
     try {
-      if ((exerciseList = JSON.parse(await AsyncStorage.getItem('E' + sectionID))) === null){
-        throw new Error('JSON parse error in getExerciseList', error);
-      }
-    } catch (e) {
+      videoUrl = await FileSystem.readAsStringAsync(lectureVideoPath + videoName + '.json');
+    } catch (e){
       if (e?.response?.data != null) {
-        throw new Error('Error in getExerciseList: ', e.response.data);
+        throw e.response.data;
```
Comment (ITPini) [inline eml/services/StorageService.js:359]:
```suggestion
        throw new Error('Error in getExerciseList: ', e.response.data);
```

File: eml/services/StorageService.js [hunk 4]
```diff
@@ -325,65 +333,36 @@ export const fetchLectureImage = async (imageID, lectureID) => {
...
       } else {
-        throw new Error('Error in getExerciseList: ', e);
+        throw e;
```
Comment (ITPini) [inline eml/services/StorageService.js:361]: throw new Error('Error in getExerciseList: ', e);

File: eml/services/StorageService.js [hunk 5]
```diff
@@ -325,65 +333,36 @@ export const fetchLectureImage = async (imageID, lectureID) => {
...
+  if (!resolution){
+    resolution = '360';
+  }
   try {
     if (isOnline) {
-      exerciseList = await api.getExercisesInSection(sectionID);
+      videoUrl = api.getVideoStreamUrl(videoName, resolution);
     } else {
-      throw new Error('No internet connection in getExerciseList');
+      throw new Error('No internet connection.');
```
Comment (ITPini) [inline eml/services/StorageService.js:351]:
```suggestion
      throw new Error('No internet connection in getVideoUrl');
```

File: eml/services/StorageService.js [hunk 6]
```diff
@@ -231,68 +235,72 @@ export const refreshSectionList = async (sectionList) => {
...
-        throw new Error('Error in getLectureList: ', e.response.data);
+        throw new Error('Error in getComponentsList: ', e.response.data);
       } else {
-        throw new Error('Error in getLectureList: ', e);
+        throw new Error('Error in getComponentsList: ', e);
       }
     }
   } finally {
-    return await lectureFittingModel(lectureList);
+    return componentList; //await componentFittingModel(componentList);
```
Comment (ITPini) [inline eml/services/StorageService.js:268]:
```suggestion
    return componentList;
```

**Review by ITPini — APPROVED** [review]

**Comment by sofiiagran** [comment]
DU ER FUCKING SEJ STEFAN !!!!

**Pull request merged**